target/linux/qualcommax/ipq807x/base-files/etc/board.d/02_network

```diff
@@ -55,9 +54,32 @@ ipq807x_setup_interfaces()
 	esac
 }
 
+ipq807x_setup_macs()
+{
+	local board="$1"
+	local lan_mac=""
+	local wan_mac=""
+	local label_mac=""
+
+	case "$board" in
+		linksys,mx5300)
+			label_mac=$(mtd_get_mac_ascii devinfo hw_mac_addr)
+			for i in $(seq 1 4); do
+				[ "$(mtd_get_mac_ascii u_env eth${i}addr)" != "$label_mac" ] && lan_mac=$label_mac
```

Comment:
@robimarko What do you think about this implementation of MAC address setting as setting from DTS is not possible?

Reply:
Why is it not possible?

Does U-Boot env have `ethaddr` variable?
If so, you can then just increment all other MAC-s via it.

Reply:
On some devices `ethaddr`, `eth1addr`, ... are different than `hw_mac_addr` from `devinfo` partition.

Reply:
Ugh, I remember now that the U-Boot parser is not yet converted to NVMEM layout and it has the `ethaddr` variable hardcoded for MAC, so you cannot parse custom variable names.

If U-Boot sees different MAC-s, I suppose it then uses the "wrong" MAC-s for patching the DTS as well?

Reply:
> If U-Boot sees different MAC-s, I suppose it then uses the "wrong" MAC-s for patching the DTS as well?

Yes, MAC addresses are incorrect.
For example Linksys MX42CF (branded MX4200) has:
- `hw_mac_addr`: `80:69:1A:xx:xx:xx`
- `ethaddr`: `E8:9F:80:xx:xx:xx`
- `caldata`: `00:03:7F:xx:xx:xx`

Reply:
For my router, the MAC addresses are correct in all three places. Is it worth implementing MAC address setting? I don't know if there is any branded version of the MX5300.

Reply:
I am not following, if they are different in those 2 places, which one is correct?

Reply:
Correct one is `hw_mac_addr` from `devinfo` partition.

Reply:
I removed the MAC address changes completely. There is currently no known MX5300 with incorrectly configured addresses.